pymeasure/instruments/keysight/keysightDSOX1102G.py

```diff
@@ -0,0 +1,559 @@
+        "window", "xy", or "roll".""",
+        validator=strict_discrete_set,
+        values={"main": "MAIN", "window": "WIND", "xy": "XY", "roll": "ROLL"},
+        map_values=True
+    )
+
+    timebase_offset = Instrument.control(
+        ":TIMebase:POSition?", ":TIMebase:REFerence CENTer;:TIMebase:POSition %f",
+        """ A float parameter that sets the time interval in seconds between the trigger 
+        event and the reference position (at center of screen by default)."""
+    )
+
+    timebase_range = Instrument.control(
+        ":TIMebase:RANGe?", ":TIMebase:RANGe %f",
+        """ A float parameter that sets the full-scale horizontal time in seconds for the 
+        main window."""
+    )
+
+    timebase_scale = Instrument.control(
+        ":TIMebase:SCALe?", ":TIMebase:SCALe %f",
+        """ A float parameter that sets the horizontal scale (units per division) in seconds 
+        for the main window."""
+    )
+
+    #################
+    # Trigger setup #
+    #################
+
+    # TODO: Implement Trigger functionality
+
+    ###############
+    # Acquisition #
+    ###############
+
+    acquisition_type = Instrument.control(
+        ":ACQuire:TYPE?", ":ACQuire:TYPE %s",
+        """ A string parameter that sets the type of data acquisition. Can be "normal", "average",
+        "hresolution", or "peak".""",
+        validator=strict_discrete_set,
+        values={"normal": "NORM", "average": "AVER", "hresolution": "HRES", "peak": "PEAK"},
+        map_values=True
+    )
+
+    acquisition_mode = Instrument.control(
+        ":ACQuire:MODE?", ":ACQuire:MODE %s",
+        """ A string parameter that sets the acquisition mode. Can be "realtime" or "segmented".""",
+        validator=strict_discrete_set,
+        values={"realtime": "RTIM", "segmented": "SEGM"},
+        map_values=True
+    )
+
+    def run(self):
+        """ Starts repetitive acquisitions. This is the same as pressing the Run key on the front panel."""
+        self.write(":run")
+
+    def stop(self):
+        """  Stops the acquisition. This is the same as pressing the Stop key on the front panel."""
+        self.write(":stop")
+
+    def single(self):
+        """ Causes the instrument to acquire a single trigger of data.
+        This is the same as pressing the Single key on the front panel. """
+        self.write(":single")
+
+    _digitize = Instrument.setting(
+        ":DIGitize %s",
+        """ Acquire waveforms according to the settings of the :ACQuire commands and specified source,
+         as a string parameter that can take the following values: "channel1", "channel2", "function",
+         "math", "fft", "abus", or "ext". """,
+        validator=strict_discrete_set,
+        values={"channel1": "CHAN1", "channel2": "CHAN2", "function": "FUNC", "math": "MATH",
+                "fft": "FFT", "abus": "ABUS", "ext": "EXT"},
+        map_values=True
+    )
+
+    def digitize(self, source: str):
```

Comment:
Digitize is not actually a property - it is an operation. I thought it made more sense from a usability point-of-view to call it like a method, with digitize(param), as opposed to using digitize = param. Furthermore, digitize cannot be read (i.e. this would not work: var = digitize).

I define _digitize as Instrument.setting simply to facilitate the verification of correct values.

Reply:
Fair enough, thanks.